google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/stub/metrics/MetricsTracerTest.java

```diff
@@ -0,0 +1,401 @@
+
+    Stopwatch stopwatch = Stopwatch.createStarted();
+    Lists.newArrayList(stub.readRowsCallable().call(Query.create(TABLE_ID)));
+    long elapsed = stopwatch.elapsed(TimeUnit.MILLISECONDS);
+
+    // Give OpenCensus a chance to update the views asynchronously.
+    Thread.sleep(100);
+
+    long opLatency =
+        getAggregationValueAsLong(
+            RpcViewConstants.BIGTABLE_OP_LATENCY_VIEW,
+            ImmutableMap.of(
+                RpcMeasureConstants.BIGTABLE_OP, TagValue.create("Bigtable.ReadRows"),
+                RpcMeasureConstants.BIGTABLE_STATUS, TagValue.create("OK")));
+    assertThat(opLatency).isIn(Range.closed(sleepTime, elapsed));
+  }
+
+  @Test
+  public void testReadRowsOpCount() throws InterruptedException {
+    doAnswer(
+            new Answer() {
+              @Override
+              public Object answer(InvocationOnMock invocation) {
+                @SuppressWarnings("unchecked")
+                StreamObserver<ReadRowsResponse> observer =
+                    (StreamObserver<ReadRowsResponse>) invocation.getArguments()[1];
+                observer.onNext(DEFAULT_READ_ROWS_RESPONSES);
+                observer.onCompleted();
+                return null;
+              }
+            })
+        .when(mockService)
+        .readRows(any(ReadRowsRequest.class), anyObserver(ReadRowsResponse.class));
+
+    Lists.newArrayList(stub.readRowsCallable().call(Query.create(TABLE_ID)));
+    Lists.newArrayList(stub.readRowsCallable().call(Query.create(TABLE_ID)));
+
+    // Give OpenCensus a chance to update the views asynchronously.
+    Thread.sleep(100);
+
+    long opLatency =
+        getAggregationValueAsLong(
+            RpcViewConstants.BIGTABLE_COMPLETED_OP_VIEW,
+            ImmutableMap.of(
+                RpcMeasureConstants.BIGTABLE_OP, TagValue.create("Bigtable.ReadRows"),
+                RpcMeasureConstants.BIGTABLE_STATUS, TagValue.create("OK")));
+    assertThat(opLatency).isEqualTo(2);
+  }
+
+  @Test
+  public void testReadRowsFirstRow() throws InterruptedException {
+    final long beforeSleep = 50;
+    final long afterSleep = 50;
+
+    doAnswer(
+            new Answer() {
+              @Override
+              public Object answer(InvocationOnMock invocation) throws Throwable {
+                @SuppressWarnings("unchecked")
+                StreamObserver<ReadRowsResponse> observer =
+                    (StreamObserver<ReadRowsResponse>) invocation.getArguments()[1];
+                Thread.sleep(beforeSleep);
+                observer.onNext(DEFAULT_READ_ROWS_RESPONSES);
+                Thread.sleep(afterSleep);
+                observer.onCompleted();
+                return null;
+              }
+            })
+        .when(mockService)
+        .readRows(any(ReadRowsRequest.class), anyObserver(ReadRowsResponse.class));
+
+    Stopwatch stopwatch = Stopwatch.createStarted();
+    Lists.newArrayList(stub.readRowsCallable().call(Query.create(TABLE_ID)));
```

Comment:
why are we wrapping this in a list?

Reply:
To read the stream...the returned iterator is lazy and needs to be consumed